python/src/tstools/inputFileIO.py

```diff
@@ -74,57 +76,57 @@ def readFitFile(self, fileName):
-                    self.dc[2] = splitLine[3]
+                    self.dc[0] = float(splitLine[1])
+                    self.dc[1] = float(splitLine[2])
+                    self.dc[2] = float(splitLine[3])
 
                 elif flag == 'VE:':
 
-                    self.ve[0] = splitLine[1]
-                    self.ve[1] = splitLine[2]
-                    self.ve[2] = splitLine[3]
+                    self.ve[0] = float(splitLine[1])
+                    self.ve[1] = float(splitLine[2])
+                    self.ve[2] = float(splitLine[3])
 
                 elif flag == 'SA:':
 
-                    self.sa[0] = splitLine[1]
-                    self.sa[1] = splitLine[2]
-                    self.sa[2] = splitLine[3]
+                    self.sa[0] = float(splitLine[1])
+                    self.sa[1] = float(splitLine[2])
+                    self.sa[2] = float(splitLine[3])
                 
                 elif flag == 'CA:':
 
-                    self.ca[0] = splitLine[1]
-                    self.ca[1] = splitLine[2]
-                    self.ca[2] = splitLine[3]
+                    self.ca[0] = float(splitLine[1])
+                    self.ca[1] = float(splitLine[2])
+                    self.ca[2] = float(splitLine[3])
 
                 elif flag == 'SS:':
 
-                    self.ss[0] = splitLine[1]
-                    self.ss[1] = splitLine[2]
-                    self.ss[2] = splitLine[3]
+                    self.ss[0] = float(splitLine[1])
+                    self.ss[1] = float(splitLine[2])
+                    self.ss[2] = float(splitLine[3])
                 
                 elif flag == 'CS:':
 
-                    self.cs[0] = splitLine[1]
-                    self.cs[1] = splitLine[2]
-                    self.cs[2] = splitLine[3]
+                    self.cs[0] = float(splitLine[1])
+                    self.cs[1] = float(splitLine[2])
+                    self.cs[2] = float(splitLine[3])
 
                 elif flag == 'O2:':
 
-                    self.o2[0] = splitLine[1]
-                    self.o2[1] = splitLine[2]
-                    self.o2[2] = splitLine[3]
+                    self.o2[0] = float(splitLine[1])
+                    self.o2[1] = float(splitLine[2])
+                    self.o2[2] = float(splitLine[3])
 
                 elif flag == 'O3:':
 
-                    self.o3[0] = splitLine[1]
-                    self.o3[1] = splitLine[2]
-                    self.o3[2] = splitLine[3]
+                    self.o3[0] = float(splitLine[1])
+                    self.o3[1] = float(splitLine[2])
+                    self.o3[2] = float(splitLine[3])
 
                 elif flag == 'O4:':
 
-                    self.o4[0] = splitLine[1]
-                    self.o4[1] = splitLine[2]
-                    self.o4[2] = splitLine[3]
+                    self.o4[0] = float(splitLine[1])
+                    self.o4[1] = float(splitLine[2])
+                    self.o4[2] = float(splitLine[3])
```

Comment:
Few thoughts about all this:

- You could have a function: `processInput` which does the `float` call... so you don't need to do it manually every time (and if it changes, you only have to change it in one place):

```python
def _processInput(input)
  return float(input)
```

```python
self.o4[0] = _processInput(splitLine[1])
```

- These blocks could maybe use some constants for the flags

- You could take a look at using a [switch statement](https://jaxenter.com/implement-switch-case-statement-python-138315.html) for this flow

- Each of these blocks (except the first few) is essentially doing the same thing. "Given a flag, set the 1st, 2nd, and 3rd value of an array to the 2nd, 3rd and 4th value of the input line, respectively". You could potentially make this more generic?

```python
switch(flag) {
  case IM:
    # do the thing
    break
  case LM:
    # do the thing
    break
  case RE:
   # do the thing
    break
  case DC:
    _setArrayVals(self.dc, splitLine)
    break
  case VE:
    _setArrayVals(self.ve, splitLine)
    break
}
```

```python
def _setArrayVals(array, input)
  array[0] = _processInput(input[0])
  array[1] = _processInput(input[1])
  array[2] = _processInput(input[2])
```